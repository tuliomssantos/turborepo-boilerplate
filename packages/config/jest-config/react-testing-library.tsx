import React from 'react'

import { render } from '@testing-library/react'

const customRender = (ui: React.ReactNode, options = {}) =>
  render(<React.Fragment>{ui}</React.Fragment>, {
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { customRender as render }
export { default as userEvent } from '@testing-library/user-event'
