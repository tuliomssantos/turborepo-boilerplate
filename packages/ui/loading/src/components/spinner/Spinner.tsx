import styles from './styles.module.scss'

//create props for Spinner using React.ComponentPropsWithoutRef
export type SpinnerProps = React.ComponentPropsWithoutRef<'div'>

/**
 * Spinner component
 *
 * use css variable --meta-spinner-size to change size
 *
 * @example
 * .spinner {
 *  --meta-spinner-size: 20px;
 * }
 */
const Spinner = ({ ...props }: SpinnerProps) => {
  return <div className={`${styles.spinner} ${props.className}`} {...props} />
}

export default Spinner
