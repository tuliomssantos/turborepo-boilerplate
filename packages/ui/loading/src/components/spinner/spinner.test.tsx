import { render } from 'jest-common/react-testing-library'

import Spinner from './Spinner'

describe('Spinner', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Spinner />)
    expect(asFragment()).toMatchSnapshot()
  })
})
