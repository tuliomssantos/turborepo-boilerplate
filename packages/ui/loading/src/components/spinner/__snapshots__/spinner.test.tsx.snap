// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner"
  />
</DocumentFragment>
`;
